planning/behavior_velocity_planner/src/utilization/util.cpp

```diff
@@ -593,34 +593,62 @@ LineString2d extendLine(
     {(p1 - length * t).x(), (p1 - length * t).y()}, {(p2 + length * t).x(), (p2 + length * t).y()}};
 }
 
-std::vector<lanelet::ConstLanelet> getLaneletsOnPath(
+boost::optional<int64_t> getNearestLaneId(
   const autoware_auto_planning_msgs::msg::PathWithLaneId & path,
-  const lanelet::LaneletMapPtr lanelet_map, const geometry_msgs::msg::Pose & current_pose)
+  const lanelet::LaneletMapPtr lanelet_map, const geometry_msgs::msg::Pose & current_pose,
+  boost::optional<size_t> & nearest_segment_idx)
 {
-  std::set<int64_t> unique_lane_ids;
-  auto nearest_segment_idx = tier4_autoware_utils::findNearestSegmentIndex(
+  boost::optional<int64_t> nearest_lane_id;
+
+  nearest_segment_idx = tier4_autoware_utils::findNearestSegmentIndex(
     path.points, current_pose, std::numeric_limits<double>::max(), M_PI_2);
 
-  // Add current lane id
   lanelet::ConstLanelets current_lanes;
   if (
     lanelet::utils::query::getCurrentLanelets(
       lanelet::utils::query::laneletLayer(lanelet_map), current_pose, &current_lanes) &&
     nearest_segment_idx) {
     for (const auto & ll : current_lanes) {
-      if (
-        ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0) ||
-        ll.id() == path.points.at(*nearest_segment_idx + 1).lane_ids.at(0)) {
-        unique_lane_ids.insert(ll.id());
+      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
+        *nearest_lane_id = ll.id();
```

Comment:
@tkimura4 
Can you also fix this nearest_lane_id uninitialized pointer?
L623 too

Reply:
Fixed at [b725fd3](https://github.com/autowarefoundation/autoware.universe/pull/1323/commits/b725fd34bf97f5c213fd9a13eb171ddb8ff480ac)

Reply:
you are inserting value to boost::get() method at L622
I think this line should be nearest_lane_id = ll.id();
```
#9  0x00007f74f3877c01 in behavior_velocity_planner::planning_utils::getNearestLaneId (path=..., lanelet_map=std::shared_ptr<lanelet::LaneletMap> (use count 3, weak count 0) = {...}, current_pose=..., 
    nearest_segment_idx=...) at /home/t4tanaka/workspace/humble_autoware/src/universe/autoware.universe/planning/behavior_velocity_planner/src/utilization/util.cpp:617
617	        *nearest_lane_id = ll.id();

```